### PR TITLE
#228 - Basic Hello World Slack Command Bot

### PR DESCRIPTION
**Pull request by ManasMadan**
## What does this PR do?

This PR Closes the issue #228 by implementing a Slack Command Bot Template in Node.js

## Test Plan
- I have not changed any code but have written the code in node/slack-command-bot directory.
- I have tested the code on Appwrite Cloud by creating a function and a Slack Command on Slack.
[Demo Video Link](https://youtu.be/JPO7AuHKUX0)
- I have made sure the API Verify the Request according to [Slack Documentation](https://api.slack.com/authentication/verifying-requests-from-slack)

## Related PRs and Issues
This PR is related to issue #228 and merging this PR closes that issue

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes I Have read the contributing guidelines

### CODE REVIEW

**Comment by ManasMadan** [comment]
@Haimantika Merging this PR closes issue #228

**Comment by ManasMadan** [comment]
@Haimantika any updates ?

**Comment by Haimantika** [comment]
> @Haimantika any updates ?

I have assigned my team mate. You can also join us for PR reviews tomorrow on Discord and get it reviewed live.

**Comment by ManasMadan** [comment]
@loks0n Any Updates ?

File: node/slack-command-bot/README.md
Comment (Meldiron) [inline node/slack-command-bot/README.md:1]:
```suggestion
# 🤖 Node.js Slack Command Bot Function
```

File: node/slack-command-bot/src/main.js
Comment (Meldiron) [inline node/slack-command-bot/src/main.js:58]: This is left-over from starter. If Slack doesn't require any specific response body, let's respond with just `OK`.

**Review by Meldiron — CHANGES_REQUESTED** [review]
Very solid PR ✨ I left some comments to ensure we follow our typical approach to templates. Please address and Ill give it another look

File: node/slack-command-bot/src/main.js
Comment (ManasMadan) [reply node/slack-command-bot/src/main.js:58]: Slack Needs A JSON or a message to be sent so it can be displayed in the bot

**Comment by ManasMadan** [comment]
@Meldiron I will do the changes and let you know.
Thanks !

**Comment by ManasMadan** [comment]
@Meldiron @loks0n I have did all the changes and have tested them Please check
Thanks !

File: node/slack-command-bot/src/main.js
Comment (loks0n) [inline node/slack-command-bot/src/main.js:17]: Can we remove this comment?

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:28]: Can this function just accept req as the argument and access the headers inside?

File: node/slack-command-bot/src/main.js
Comment (loks0n) [inline node/slack-command-bot/src/main.js:10]: Seems like these functions are basically of the same purpose, can we put the replay check inside the throwIfRequestNotVerified

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:34]: Can we use sentence case in the error message?

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:48]: Can we use sentence case in the error message?

File: node/slack-command-bot/README.md
Comment (loks0n) [inline node/slack-command-bot/README.md:9]: Can we use sentence case in this description

File: node/slack-command-bot/README.md
Comment (loks0n) [inline node/slack-command-bot/README.md:41]: Sentence case here please

**Review by loks0n — CHANGES_REQUESTED** [review]

**Comment by ManasMadan** [comment]
@loks0n Have done the changes
Thanks !

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:41]: This error function does not exist in this scope. You can remove this line.

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:32]: This error function does not exist in this scope. You can remove this line.

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:33]: I think `Invalid request: replay attack` looks better

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:42]: I think `Invalid request: incorrect signature` looks better

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:36]: Can we use a template literal string here?

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:39]: Can we use a template literal string here?

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:28]: I don't think this inline var is necessary.

Better to just use req.bodyRaw when needed

**Review by loks0n — CHANGES_REQUESTED** [review]

**Comment by ManasMadan** [comment]
@loks0n Please check now have done the changes
Thanks !

**Comment by ManasMadan** [comment]
@loks0n @Meldiron any updates ?

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:37]: This name isn't the best, how about `expectedSignature`, `validSignature` or `correctSignature`?

**Comment by ManasMadan** [comment]
@loks0n have done the changes
Please Check
Thanks !

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:38]: Can we change this to `expectedSignature !== signature`?

File: node/slack-command-bot/src/utils.js
Comment (loks0n) [inline node/slack-command-bot/src/utils.js:36]: Can we add a newline below for readability?

**Review by loks0n — APPROVED** [review]

**Review by loks0n — APPROVED** [review]

**Comment by ManasMadan** [comment]
@Haimantika Am i Eligible for Appwrite Swags ?
If Yes do i need to do something to get them

**Comment by ManasMadan** [comment]
@Meldiron @loks0n can you please merge this

**Comment by ManasMadan** [comment]
> @Haimantika Am i Eligible for Appwrite Swags ? If Yes do i need to do something to get them

@Haimantika

**Comment by loks0n** [comment]
> @Haimantika Am i Eligible for Appwrite Swags ? If Yes do i need to do something to get them

Don't worry - you are eligible for the swags 🥳 

We will merge the PR when Meldiron is able to review.

**Comment by tessamero** [comment]
@ManasMadan add your discord handle here so we can contact you about swags after the PR is merged soon :)

**Comment by ManasMadan** [comment]
@tessamero My discord username is manasmadan

**Review by Meldiron — APPROVED** [review]

**Pull request merged**
